webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-sort-name.js

```diff
@@ -0,0 +1,44 @@
+import {RubricsElement} from "./rubrics-element.js";
+import {html} from "/webcomponents/assets/lit-element/lit-element.js";
+
+export class SakaiRubricSortName extends RubricsElement {
+
+  constructor() {
+
+    super();
+
+    this.userId = "";
+    this.sortName = "";
+  }
+
+  static get properties() {
+    return { userId: {attribute: "user-id", type: String}, sortName: String };
+  }
+
+  attributeChangedCallback(name, oldValue, newValue) {
+
+    super.attributeChangedCallback(name, oldValue, newValue);
+
+    if (name === "user-id") {
+      this.setSortName();
+    }
+  }
+
+  render() {
+    return html`${this.sortName}`;
+  }
+
+  setSortName() {
+
+    var self = this;
+    jQuery.ajax({
+      url: '/sakai-ws/rest/sakai/getUserSortName?eid=' + this.userId
```

Comment:
It will, the way it currently works.
I've been trying to rectify this for days and can't seem to figure out a way that works consistently. Basically what I'm going for is a way to gather a list of all the user IDs from the evaluations, do a single Ajax request that returns them along with their corresponding sort names, and then just replace the IDs with the sort names based on the data from that single request. Did you have a different pattern in mind to follow, or does this sound like a valid path? I've tried a ton of slightly different versions of this approach and haven't been able to make one work yet. I'd love suggestions if anyone has them!

Reply:
Closest I've gotten involved making the Ajax request synchronous, which didn't seem to be such a great idea in terms of speed.